### PR TITLE
Support SPV_KHR_fragment_shading_rate

### PR DESCRIPTION
**Pull request by Tobski**
Relies on https://github.com/KhronosGroup/SPIRV-Headers/pull/172

### CODE REVIEW

**Review by alan-baker — APPROVED** [review]
Looks good, will wait till headers are merged to re-trigger the bots.

**Comment by Tobski** [comment]
Checks are all now passing and has approval from @alan-baker - anything else we need to get this merged?

File: source/val/validate_builtins.cpp
Comment (SpaceIm) [inline source/val/validate_builtins.cpp:3676]: Does it compile? I guess these 2 `case` should be outside the block of `case SpvBuiltInRayGeometryIndexKHR`.

**Pull request merged**
